compiler/rustc_resolve/src/imports.rs

```diff
@@ -990,7 +990,10 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                         });
                         let res = binding.res();
                         if let Ok(initial_res) = initial_res {
-                            if res != initial_res && this.ambiguity_errors.is_empty() {
+                            if res != initial_res
+                                && this.ambiguity_errors.is_empty()
+                                && res != Res::Err
+                            {
```

Comment:
It seems that we can extract the common logic:

```rs
if res == Res::Err || !this.ambiguity_errors.is_empty() {
    return;
}
if let Ok(initial_res) = initial_res {
    if res != initial_res {
        span_bug!(import.span, "inconsistent resolution for an import");
    }
} else if this.privacy_errors.is_empty() {
    this.tcx
    .sess
    .create_err(CannotDetermineImportResolution { span: import.span })
    .emit();
}
```

Reply:
If doing that, before the `return;` we'd want to have a `delay_span_bug` call to guard against miscompilations, while also silencing other errors have occurred (as is expected).

Reply:
Yes, makes sense.